modules/oadp-gcp-wif-cloud-authentication.adoc

```diff
@@ -0,0 +1,95 @@
+
+WIF handles encrypting and decrypting certificates, extracting user attributes, and validation. WIF externalizes authentication, passing it over to Security Token Services (STS) and reduces the demands on individual developers. Authorization and controlling access to resources remain the responsibility of the application.
+
+[NOTE]
+====
+Google WIF cloud authentication is avail for OADP 1.3.x and later.
+====
+
+If you do not use Google WIF cloud authentication, continue to xref:../../..//backup_and_restore/application_backup_and_restore/installing/oadp-installing-operator.adoc#oadp-installing-operator-doc[Installing the OADP Operator].
+
+.Prerequisites
+
+* You must install a cluster in manual mode with link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.13/html-single/authentication_and_authorization/index#gcp-workload-identity-mode-installing[GCP Workload Identity configured].
+* You now have access to `ccoctl` CLI and access to the associated workload-identity-pool.
+
+.Procedure
+
+. Create an oadp-credrequest directory.
++
+[source,terminal]
+----
+$ mkdir -p oadp-credrequest
+----
++
+. Create a credrequest.yaml.
++
+[source,yaml]
+----
+echo 'apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: oadp-operator-credentials
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: GCPProviderSpec
+    permissions:
+    - compute.disks.get
+    - compute.disks.create
+    - compute.disks.createSnapshot
+    - compute.snapshots.get
+    - compute.snapshots.create
+    - compute.snapshots.useReadOnly
+    - compute.snapshots.delete
+    - compute.zones.get
+    - storage.objects.create
+    - storage.objects.delete
+    - storage.objects.get
+    - storage.objects.list
+    - iam.serviceAccounts.signBlob
+    skipServiceCheck: true
+  secretRef:
+    name: cloud-credentials-gcp
+    namespace: <OPERATOR_INSTALL_NS>
+  serviceAccountNames:
+  - velero
+' > oadp-credrequest/credrequest.yaml
+----
++
+. Use `ccoctl` to create the `credrequest` pointing to the directory `oadp-credrequest`
++
+[source,terminal]
+----
+ccoctl gcp create-service-accounts --name=<name> \
+    --project=<gcp-project-id> \
+    --credentials-requests-dir=oadp-credrequest \
+    --workload-identity-pool=<pool-id> \
+    --workload-identity-provider=<provider-id>
+----
++
+. Use the ccoctl reference to generate  `manifests/openshift-adp-cloud-credentials-gcp-credentials.yaml` to use in the following step.
++
+. Apply the credentials secret to the openshift-adp namespace
++
+[source,terminal]
+----
+oc create namespace openshift-adp
+oc apply -f manifests/openshift-adp-cloud-credentials-gcp-credentials.yaml
+----
++
+. Generate a token...?
```

Comment:
Do we need a section on "Generating tokens?"

Reply:
```
ccoctl gcp create-service-accounts --name=<name> \
    --project=<gcp-project-id> \
    --credentials-requests-dir=oadp-credrequest \
    --workload-identity-pool=<pool-id> \
    --workload-identity-provider=<provider-id>
```
from earlier generated the token to use already which we are applying in
```
oc apply -f manifests/openshift-adp-cloud-credentials-gcp-credentials.yaml
```

Reply:
so no.